GDJS/Runtime/RuntimeInstanceContainer.ts

```diff
@@ -102,6 +102,70 @@ namespace gdjs {
       result: FloatPoint
     ): FloatPoint;
 
+    /**
+     * @return the left bound of:
+     * - the game resolution for a {@link gdjs.RuntimeScene}
+     * - the default dimensions (the AABB of all its children) for a
+     * {@link gdjs.CustomRuntimeObject}.
+     */
+    abstract getUnrotatedViewportMinX(): float;
+
+    /**
+     * @return the top bound of:
+     * - the game resolution for a {@link gdjs.RuntimeScene}
+     * - the default dimensions (the AABB of all its children) for a
+     * {@link gdjs.CustomRuntimeObject}.
+     */
+    abstract getUnrotatedViewportMinY(): float;
+
+    /**
+     * @return the right bound of:
+     * - the game resolution for a {@link gdjs.RuntimeScene}
+     * - the default dimensions (the AABB of all its children) for a
+     * {@link gdjs.CustomRuntimeObject}.
+     */
+    abstract getUnrotatedViewportMaxX(): float;
+
+    /**
+     * @return the bottom bound of:
+     * - the game resolution for a {@link gdjs.RuntimeScene}
+     * - the default dimensions (the AABB of all its children) for a
+     * {@link gdjs.CustomRuntimeObject}.
+     */
+    abstract getUnrotatedViewportMaxY(): float;
+
+    /**
+     * @return the left bound of:
+     * - the initial game resolution for a {@link gdjs.RuntimeScene}
+     * - the initial default dimensions (inner area) set in the editor for a
+     * {@link gdjs.CustomRuntimeObject}.
+     */
+    abstract getOriginalViewportMinX(): float;
```

Comment:
As seen with you:

- Unrotated === Unscaled (it's ok)
- But "Original" vs (nothing) is unrelated to Unrotated

So it should be called "getOriginalUnrotatedViewport"
(or getOriginalUnscaledViewport)
(or anything else that is just adding "Original")